inference-engine/tests/functional/inference_engine/ir_serialization/ordering.cpp

```diff
@@ -0,0 +1,163 @@
+    const auto& attributes1 = walker1.attributes;
+    const auto& attributes2 = walker2.attributes;
+    if (attributes1.size() != attributes2.size()) {
+        return {false, "Attribute count differ: " +
+                           std::to_string(attributes1.size()) + " != " +
+                           std::to_string(attributes2.size())};
+    }
+
+    // every attribute must be equal
+    for (int i = 0; i < attributes1.size(); i++) {
+        const std::string a1_name{attributes1[i].name()};
+        const std::string a2_name{attributes2[i].name()};
+        const std::string a1_value{attributes1[i].value()};
+        const std::string a2_value{attributes2[i].value()};
+        if ((a1_name != a2_name) || (a1_value != a2_value)) {
+            return {false, "Attributes types differ in <layer> : " + a1_name +
+                               "=" + a1_value + " != " + a2_name + "=" +
+                               a2_value};
+        }
+    }
+    return {true, ""};
+}
+}  // namespace
+
+class OrderingSerializationTest : public ::testing::Test {
+protected:
+    std::string test_name =
+        ::testing::UnitTest::GetInstance()->current_test_info()->name();
+    std::string m_out_xml_path = test_name + ".xml";
+    std::string m_out_bin_path = test_name + ".bin";
+
+    void TearDown() override {
+        std::remove(m_out_xml_path.c_str());
+        std::remove(m_out_bin_path.c_str());
+    }
+};
+
+TEST_F(OrderingSerializationTest, BasicModel_MO) {
+    const std::string model = IR_SERIALIZATION_MODELS_PATH "add_abc.xml";
+    const std::string weights = IR_SERIALIZATION_MODELS_PATH "add_abc.bin";
+
+    InferenceEngine::Core ie;
+    auto m = ie.ReadNetwork(model, weights);
+    m.serialize(m_out_xml_path, m_out_bin_path);
+
+    pugi::xml_document expected_xml;
+    pugi::xml_document result_xml;
+    ASSERT_TRUE(expected_xml.load_file(model.c_str()));
+    ASSERT_TRUE(result_xml.load_file(m_out_xml_path.c_str()));
+
+    bool success;
+    std::string message;
+    std::tie(success, message) =
+        compare_layer_ordering(expected_xml, result_xml);
+    ASSERT_TRUE(success) << message;
+}
+
+TEST_F(OrderingSerializationTest, ModelWithMultipleOutputs_MO) {
+    const std::string model =
+        IR_SERIALIZATION_MODELS_PATH "split_equal_parts_2d.xml";
+    const std::string weights =
+        IR_SERIALIZATION_MODELS_PATH "split_equal_parts_2d.bin";
+
+    InferenceEngine::Core ie;
+    auto m = ie.ReadNetwork(model, weights);
+    m.serialize(m_out_xml_path, m_out_bin_path);
+
+    pugi::xml_document expected_xml;
+    pugi::xml_document result_xml;
+    ASSERT_TRUE(expected_xml.load_file(model.c_str()));
+    ASSERT_TRUE(result_xml.load_file(m_out_xml_path.c_str()));
+
+    bool success;
+    std::string message;
+    std::tie(success, message) =
+        compare_layer_ordering(expected_xml, result_xml);
+    ASSERT_TRUE(success) << message;
+}
+
+TEST_F(OrderingSerializationTest, ModelWithMultipleLayers_MO) {
```

Comment:
This test is failing, new get_ordered_ops() is producing different layer ordering. @GlebKazantaev could you help me out here? Is it possible that MO uses different algorithm in some edge cases?

[IR produced by MO](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/functional/inference_engine/ir_serialization/models/addmul_abc.xml)
```
<?xml version="1.0" ?>
<net name="addmul_abc" version="10">
	<layers>
		<layer id="0" name="A" type="Parameter" version="opset1">			
		<layer id="1" name="B" type="Parameter" version="opset1">			
		<layer id="2" name="add_node1" type="Add" version="opset1">			
		<layer id="3" name="add_node2" type="Multiply" version="opset1">			
		<layer id="4" name="add_node3" type="Add" version="opset1">			
		<layer id="5" name="C" type="Parameter" version="opset1">			
		<layer id="6" name="add_node4" type="Add" version="opset1">			
		<layer id="7" name="Y" type="Add" version="opset1">			
		<layer id="8" name="Y/sink_port_0" type="Result" version="opset1">			
	</layers>
	<edges>
		<edge from-layer="0" from-port="0" to-layer="2" to-port="0"/>
		<edge from-layer="1" from-port="0" to-layer="2" to-port="1"/>
		<edge from-layer="0" from-port="0" to-layer="3" to-port="0"/>
		<edge from-layer="1" from-port="0" to-layer="3" to-port="1"/>
		<edge from-layer="2" from-port="2" to-layer="4" to-port="0"/>
		<edge from-layer="3" from-port="2" to-layer="4" to-port="1"/>
		<edge from-layer="4" from-port="2" to-layer="6" to-port="0"/>
		<edge from-layer="5" from-port="0" to-layer="6" to-port="1"/>
		<edge from-layer="6" from-port="2" to-layer="7" to-port="0"/>
		<edge from-layer="5" from-port="0" to-layer="7" to-port="1"/>
		<edge from-layer="7" from-port="2" to-layer="8" to-port="0"/>
	</edges>	
</net>
```

IR produced by new get_ordered_ops():
```
<?xml version="1.0"?>
<net name="addmul_abc" version="10">
	<layers>
		<layer id="0" name="A" type="Parameter" version="opset1">		
		<layer id="1" name="B" type="Parameter" version="opset1">		
		<layer id="2" name="add_node1" type="Add" version="opset1">		
		<layer id="3" name="add_node2" type="Multiply" version="opset1">		
		<layer id="4" name="add_node3" type="Add" version="opset1">		
		<layer id="5" name="add_node4" type="Add" version="opset1">		
		<layer id="6" name="C" type="Parameter" version="opset1">		
		<layer id="7" name="Y" type="Add" version="opset1">		
		<layer id="8" name="Y/sink_port_0" type="Result" version="opset1">
	</layers>
	<edges>
		<edge from-layer="0" from-port="0" to-layer="2" to-port="0" />
		<edge from-layer="0" from-port="0" to-layer="3" to-port="0" />
		<edge from-layer="1" from-port="0" to-layer="2" to-port="1" />
		<edge from-layer="1" from-port="0" to-layer="3" to-port="1" />
		<edge from-layer="2" from-port="2" to-layer="4" to-port="0" />
		<edge from-layer="3" from-port="2" to-layer="4" to-port="1" />
		<edge from-layer="4" from-port="2" to-layer="5" to-port="0" />
		<edge from-layer="5" from-port="2" to-layer="7" to-port="0" />
		<edge from-layer="6" from-port="0" to-layer="5" to-port="1" />
		<edge from-layer="6" from-port="0" to-layer="7" to-port="1" />
		<edge from-layer="7" from-port="2" to-layer="8" to-port="0" />
	</edges>
</net>
```
Test result:
```
[==========] Running 4 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 4 tests from OrderingSerializationTest
[ RUN      ] OrderingSerializationTest.BasicModel_MO
[       OK ] OrderingSerializationTest.BasicModel_MO (14 ms)
[ RUN      ] OrderingSerializationTest.ModelWithMultipleOutputs_MO
/home/jdanieck/openvino/inference-engine/tests/functional/inference_engine/ir_serialization/ordering.cpp:120: Failure
Value of: success
  Actual: false
Expected: true
Attributes types differ in <layer> : type=Split != type=Const
[  FAILED  ] OrderingSerializationTest.ModelWithMultipleOutputs_MO (5 ms)
[ RUN      ] OrderingSerializationTest.ModelWithMultipleLayers_MO
/home/jdanieck/openvino/inference-engine/tests/functional/inference_engine/ir_serialization/ordering.cpp:140: Failure
Value of: success
  Actual: false
Expected: true
Attributes types differ in <layer> : type=Parameter != type=Add
[  FAILED  ] OrderingSerializationTest.ModelWithMultipleLayers_MO (5 ms)
[ RUN      ] OrderingSerializationTest.ModelWithConstants_MO
[       OK ] OrderingSerializationTest.ModelWithConstants_MO (3 ms)
[----------] 4 tests from OrderingSerializationTest (27 ms total)

[----------] Global test environment tear-down
[==========] 4 tests from 1 test case ran. (27 ms total)
[  PASSED  ] 2 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] OrderingSerializationTest.ModelWithMultipleOutputs_MO
[  FAILED  ] OrderingSerializationTest.ModelWithMultipleLayers_MO

 2 FAILED TESTS
```

Reply:
It was agreed in offline discussion that we won't spend more time on ModelWithMultipleLayers_MO test and test will be removed.